### PR TITLE
  This handler will serve a 302 response to a client request for a SP…

### PR DESCRIPTION
**Pull request by sticilface**
Serves a 302 with location if the request comes from the STA side of the ESP network. If the request comes from the AP side then it serves the file from SPIFFS.

### CODE REVIEW

File: src/FileFallbackHandler.h [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+      server.addHandler( new FileFallbackHandler(SPIFFS, "/jquery/jqm1.4.5.css", "/jquery/jqm1.4.5.css", "http://code.jquery.com/mobile/1.4.5/jquery.mobile-1.4.5.min.css",  "max-age=86400"));
+
+
+*/
+#ifndef ASYNC_FILEFALLBACK_H_
+#define ASYNC_FILEFALLBACK_H_
+
+#include <ESPAsyncWebServer.h>
+#include <ESP8266WiFi.h>
```
Comment (me-no-dev) [inline src/FileFallbackHandler.h]: this depends on the platform ;)
```cpp
#if defined(ESP31B)
#include <ESP31BWiFi.h>
#elif defined(ESP8266)
#include <ESP8266WiFi.h>
#else
#error Platform not supported
#endif

File: src/FileFallbackHandler.h [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        _path = _path.substring(0, _path.length() - 1); 
+        DEBUGF("[FileFallbackHandler] _path / removed\n"); 
+      }
+    }
+    bool canHandle(AsyncWebServerRequest *request);
+    void handleRequest(AsyncWebServerRequest *request);
+    
+};
+
+#endif
```
Comment (me-no-dev) [inline src/FileFallbackHandler.h]: add trailing new line

File: src/FileFallbackHandler.cpp [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+			response->addHeader("Cache-Control", _cache_header);
+		}
+		request->send(response);
+	} else {
+		request->send(404);
+	}
+
+	path = String();
+
+}
```
Comment (me-no-dev) [inline src/FileFallbackHandler.cpp]: add trailing new line

**Comment by andig** [comment]
I love the idea. Have manually built this on the JS side now but this is the _much_ better solution. Looking forward to get it merged :)

**Pull request merged**